### DISCUSSION TITLE
Is the refetch function referentially stable?

### DISCUSSION THREAD

**Discussion by danilo-89** [TanStack/query#7929] [Q&A] [answered]
I couldn't find the clear answer in the documentation, but it seems to me that refetch is referentially stable.
Am I right to assume so?

For example, this should be safe to do?
```
  const {
    data,
    refetch,
  } = useQuery({
    queryKey: ["test"],
    queryFn: doSomethingAsync,
    staleTime: Infinity,
    enabled: false,
  });
  
  useEffect(() => {
    refetch();
  }, [refetch]);
```

The docs say: 
> The top level object returned from useQuery, useInfiniteQuery, useMutation and the Array returned from useQueries is not referentially stable. It will be a new reference on every render.
> However, the data properties returned from these hooks will be as stable as possible.
https://tanstack.com/query/latest/docs/framework/react/guides/render-optimizations

**Comment by TkDodo** [comment, marked as answer]
yes
